src/main/java/javaslang/collection/ScanImpl.java

```diff
@@ -0,0 +1,49 @@
+package javaslang.collection;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+class ScanImpl {
```

Comment:
We should move this functionality to `javaslang.collection.Iterator` and remove the `ScanImpl` class. Whe then remove the `thiz` arg and just use `this` instead. The other classes call `iterator().scan...` instead of `ScanImpl.scan...`.